src/controllers/user.js

```diff
@@ -1,63 +1,71 @@
-import {
-  addUser,
-  getUser,
-  deleteUser,
-  updateUser,
-  getAutoSuggestUsers,
-} from '../services/user.js';
-
-function onAddUser(req, res) {
-  addUser(req.body);
-  res
-    .status(200)
-    .json({ message: `User with login = ${req.body.login} is created` });
-}
+import UserService from '../services/user.js';
 
-function onUpdateUser(req, res) {
-  const isUpdated = updateUser(req.params.id, req.body);
-  if (isUpdated) {
-    res
-      .status(200)
-      .json({ message: `User with id = ${req.params.id} is updated` });
-  } else {
-    res
-      .status(404)
-      .json({ message: `User with id = ${req.params.id} not found` });
+export default class UserController {
+  static async onAddUser(req, res, next) {
+    try {
+      res.json(await UserService.addUser(req.body));
+    } catch (err) {
+      next(err);
+    }
   }
-}
 
-function onGetUser(req, res) {
-  const user = getUser(req.params.id);
-  if (user) {
-    res.json(user);
-  } else {
-    res
-      .status(404)
-      .json({ message: `User with id = ${req.params.id} not found` });
+  static async onUpdateUser(req, res, next) {
+    try {
+      const isUpdated = await UserService.updateUser(req.params.id, req.body);
+
+      if (isUpdated) {
+        res
+          .status(200)
+          .json({ message: `User with id = ${req.params.id} is updated` });
+      } else {
+        res
+          .status(404)
+          .json({ message: `User with id = ${req.params.id} not found` });
+      }
+    } catch (err) {
+      next(err);
+    }
   }
-}
 
-function onDeleteUser(req, res) {
-  const isDeleted = deleteUser(req.params.id);
-  if (isDeleted) {
-    res
-      .status(200)
-      .json({ message: `User with id = ${req.params.id} is deleted` });
-  } else {
-    res
-      .status(404)
-      .json({ message: `User with id = ${req.params.id} not found` });
+  static async onGetUser(req, res, next) {
+    try {
+      const user = await UserService.getUser(req.params.id);
+
+      if (user) {
+        res.json(user);
+      } else {
+        res
+          .status(404)
+          .json({ message: `User with id = ${req.params.id} not found` });
+      }
+    } catch (err) {
+      next(err);
+    }
   }
-}
 
-function onGetAutosuggestedUsers(req, res) {
-  res.json(getAutoSuggestUsers(req.query));
-}
+  static async onDeleteUser(req, res, next) {
+    try {
+      const isDeleted = await UserService.deleteUser(req.params.id);
+
+      if (isDeleted) {
+        res
+          .status(200)
+          .json({ message: `User with id = ${req.params.id} is deleted` });
+      } else {
+        res
```

Comment:
I think it is not the best way to determine if user is not found. User service logic may be much more complicated and return `false` in many different usecases. The better is to check if user exists (and is not already deleted) in service layer, if no -throw an error which will be handled in error middleware, otherwise delete.

Reply:
Done